src/main/java/com/thealgorithms/backtracking/ArrayCombination.java

```diff
@@ -18,16 +19,16 @@ private ArrayCombination() {
      * @return A list of all combinations of length k.
      */
     public static List<List<Integer>> combination(int n, int k) {
-        if (n < 0 || k < 0 || k > n) {
+        if (k < 0 || k > n) {
```

Comment:
The n < 0 check is important because it ensures that the input values are valid within the context of combinations, prevents undefined behavior.

Reply:
It looks like if you remove this check everything will work correctly. And you're right, we can remove it. But this check provides readable constraints and with this check the code becomes more readable.